### DISCUSSION TITLE
Option to hide audio device

### DISCUSSION THREAD

**Discussion by krisu5** [File-New-Project/EarTrumpet#804] [Ideas]
_At least on the volume menu._

I have connected TV to my PC and I never need to change its volume settings, so that's why.

**Comment by terrehbyte** [comment]
I did a barebones implementation of this for myself on a fork, but it's nowhere near ready to submit as a PR.

https://github.com/terrehbyte/EarTrumpet

![A screenshot of the filter menu, prompting users to provide each device as lines on a text box](https://user-images.githubusercontent.com/4968773/227427923-f2b6b150-26c8-4a64-be1a-bcdac28c0e0c.png)

Currently, a new section is added to the Settings window that a series of toggles for showing/hiding devices. All devices are enabled by default, reflecting the current behavior. This only applies to the flyout window at the moment.

Some TODOs on my part:
- [x] Refresh the list if the list of devices to hide has changed (currently requires app restart)
  - [x] Fix duplicate device entries
- [x] Add a more user-friendly way of hiding devices (maybe a list of toggles?)
- [x] Find a less brittle way to track devices (does Windows have a GUID for each device?)
    - Could be possible with `IStreamWithVolumeControl.Id` which is stable between boots (but might change on a Windows update, so fuzzy matching might be nice as a fallback)
    - Updated Windows; the `Id` prop definitely doesn't seem stable across updates, ideally would implement something
- [ ] Add an option to apply these filter options to the picker window as well
- [ ] Figure out how glyphs work to change settings icon
- [x] Contact maintainers
- [ ] Always show active device even if it is normally filtered from the list

**Comment by Hill-98** [comment]
This is a great feature and I was looking for it too. Since I have two monitors connected and some speakers and headphones, I just want to simply hide some instead of disabling them.

**Comment by lumpycustard** [comment]
Sad to see this suggestion has not received any attention since 2021. I have way too many devices that i can't disable else i would break functionality of their associated apps. I just want to hide them from EarTrumpet to make swapping between outputs faster and easier.

![image](https://user-images.githubusercontent.com/19260650/224601304-9a7ec7db-f12b-4f52-b3fb-83dce292c8fb.png)

**Comment by mohcow** [comment]
This is such a must-have feature for people with a software audio mixer(voicemeeter) like me. flyout is so absurdly huge due to so many devices i can't disable. I really hope this gets implemented.

**Comment by Lhorzz** [comment]
It'd be nice if there was a feature to show only the active audio device in the right click > Volume Mixer window. I don't need to see all the other devices on it that aren't currently active. I saw someone implement a filter, but that's only with the icon left clicking.
![image](https://github.com/File-New-Project/EarTrumpet/assets/11053752/f08293cc-6328-42fb-a234-dbedf029a047)
So in this image it'd only show my speaker and not the headphone one.

**Reply by terrehbyte** [reply]
Oh, good point. I didn't even know that there was another Volume Mixer until you mentioned it. 👀

My dev build actually has three different ones...

![image](https://github.com/File-New-Project/EarTrumpet/assets/4968773/c8334372-325c-45ee-9c5e-2e3e9ae33912)

Though I think the one you're referring to is the one that's _not_ nested under Windows Legacy.

I'd imagine that there's nothing I can do about Windows' mixers, but I'll see if I can apply it to EarTrumpet's in my fork and reach out to maintainers again about merging.

**Reply by Lhorzz** [reply]
No, it's not the legacy one, it's that bottom option on your picture. I use it a lot since I have no other way to control the volume on my headset, and I have to change it a lot, so I just keep the window open on a side monitor. But would love to be able to only see what device is currently active so I don't see a long bar.

**Comment by zdimension** [comment]
This feature would be really nice. I've got lots of output devices so it's unusable (screenshot only shows half):

![image](https://github.com/File-New-Project/EarTrumpet/assets/4533568/60edc52c-2fca-4c10-87cd-221fe232c2b4)
